### PR TITLE
test(cypress): Adds end to end Cypress tests

### PR DESCRIPTION
**Pull request by justin-bits**

### CODE REVIEW

**Comment by justin-bits** [comment]
I'm getting linter errors under Travis that I wasn't getting running locally.  Is Travis using the latest version of `eslint-config-opensphere`?

`Expected space or tab after '//' in comment  spaced-comment` will be resolved by this: https://github.com/ngageoint/eslint-config-opensphere/pull/2

**Comment by justin-bits** [comment]
- Comma between folders to be removed: `    "lint:js": "eslint --cache 'src/**/*.js', 'cypress/**/*.js'",`
- Need to re-run without --cache as `npx eslint 'cypress/**/*.js'` locally

**Comment by justin-bits** [comment]
Rebuilt, but still the same linter failures.  It appears a new release will need to be created to pickup the changes. https://github.com/ngageoint/eslint-config-opensphere @wallw-bits

File: cypress/integration/smoke-tests/load-data-server-arcgis.spec.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  before('Login', function() {
+    cy.login();
+  });
+
+  it('Load data from ARCGIS server', function() {
+    // Add a server
+    cy.get(os.statusBar.SERVERS_BUTTON).click();
+    cy.get(os.settingsDialog.Panel.dataServers.ADD_SERVER_BUTTON).click();
+    cy.get(os.importURLDialog.ENTER_A_URL_FIELD)
+        .type('https://ags.auroragov.org/aurora/rest/services/OpenData/MapServer');
```
Comment (wallw-bits) [inline cypress/integration/smoke-tests/load-data-server-arcgis.spec.js]: Depending on third-party servers for tests is not going to be consistent.

File: cypress/integration/smoke-tests/load-data-server-geoserver.spec.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  before('Login', function() {
+    cy.login();
+  });
+
+  it('Load data from GeoServer', function() {
+    // Add a server
+    cy.get(os.statusBar.SERVERS_BUTTON).click();
+    cy.get(os.settingsDialog.Panel.dataServers.ADD_SERVER_BUTTON).click();
+    cy.get(os.importURLDialog.ENTER_A_URL_FIELD)
+        .type('https://gdp-geoserver.dev.dev.east.paas.geointservices.io/geoserver/ows');
```
Comment (wallw-bits) [inline cypress/integration/smoke-tests/load-data-server-geoserver.spec.js]: Depending on third-party servers for tests is not going to be consistent.

File: cypress/fixtures/smoke-tests/load-state-file-test-state_state.xml [hunk 3]
```diff
@@ -0,0 +1 @@
...922,39.75605766958 -104.80108188783922,39.68512152332343 -104.84757476992185,39.68512152332343</kml:coordinates></kml:LinearRing></kml:outerBoundaryIs></kml:Polygon><layer>state#ghn7ebhnrjq2-depdqf9ttkbk-3769575637#features</layer></queryArea></queryAreas><queryEntries><queryEntry layerId="state#ghn7ebhnrjq2-depdqf9ttkbk-1904757301#features" areaId="state#ghn7ebhnrjq2-area_vav6hrp4uoq3" filterId="state#ghn7ebhnrjq2-fuz6f0gdzjvv" includeArea="false" filterGroup="true"/><queryEntry layerId="state#ghn7ebhnrjq2-depdqf9ttkbk-1904757301#features" areaId="state#ghn7ebhnrjq2-area_vt78stc771bb" filterId="state#ghn7ebhnrjq2-fuz6f0gdzjvv" includeArea="true" filterGroup="true"/><queryEntry layerId="state#ghn7ebhnrjq2-depdqf9ttkbk-3769575637#features" areaId="state#ghn7ebhnrjq2-area_jj3494tspi0n" includeArea="false" filterGroup="true"/><queryEntry layerId="state#ghn7ebhnrjq2-depdqf9ttkbk-3769575637#features" areaId="state#ghn7ebhnrjq2-area_1k3s4yjthpe9" includeArea="true" filterGroup="true"/></queryEntries><time><interval>2019-01-07T00:00:00Z/2019-01-08T00:00:00Z</interval><current>2019-01-07T00:00:00Z/2019-01-08T00:00:00Z</current><advance>PT30M</advance><duration>day</duration></time></state>
```
Comment (wallw-bits) [inline cypress/fixtures/smoke-tests/load-state-file-test-state_state.xml]: Depending on third-party servers for tests is not going to be consistent.

**Comment by justin-bits** [comment]
Tests should kill http-server on failure.

**Comment by justin-bits** [comment]
- Setup docker container, future
- Currently issues with test:prep consistently creating a backup of settings.json, needs investigation
- Documentation remains to be done with this PR
- Tests that fail currently will not kill http-server and won't restore the settings.json

File: .gitignore [hunk 4]
```diff
@@ -44,3 +44,14 @@ Thumbs.db
...
+
+# cypress generated files #
+###########################
+cypress/videos/
+cypress/screenshots/
+
+
+# VS Code generated files #
+###########################
+.vscode/
```
[the remark this answers is not in the document]
Comment (justin-bits) [reply .gitignore]: Removed.

File: cypress/integration/smoke-tests/load-data-server-arcgis.spec.js (hunk 1 above)
Comment (justin-bits) [reply cypress/integration/smoke-tests/load-data-server-arcgis.spec.js]: https://github.com/ngageoint/opensphere/issues/398

File: cypress/integration/smoke-tests/load-data-server-geoserver.spec.js (hunk 2 above)
Comment (justin-bits) [reply cypress/integration/smoke-tests/load-data-server-geoserver.spec.js]: https://github.com/ngageoint/opensphere/issues/398

File: cypress/fixtures/smoke-tests/load-state-file-test-state_state.xml (hunk 3 above)
Comment (justin-bits) [reply cypress/fixtures/smoke-tests/load-state-file-test-state_state.xml]: https://github.com/ngageoint/opensphere/issues/398

**Pull request merged**